src/structures/Message.js

```diff
@@ -409,7 +409,7 @@ class Message extends Base {
     return (
       !this.deleted &&
       (this.author.id === this.client.user.id ||
-        (this.guild && this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false)))
+        (this.guild && this.channel.permissionsFor(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES, false)))
```

Comment:
this will return `undefined` if there are no permissions returned by `permissionsFor`, or alternatively return `null` if there is no guild, instead of a boolean like its meant to